### PR TITLE
Refactor authorized keys to use metadata package

### PR DESCRIPTION
**Pull request by ChaitanyaKulkarni28**

### CODE REVIEW

File: google_authorized_keys/main.go [hunk 1]
```diff
@@ -20,23 +20,24 @@ import (
...
 	"github.com/GoogleCloudPlatform/guest-logging-go/logger"
 )
 
-var (
-	programName       = "GoogleAuthorizedKeysCommand"
-	metadataURL       = "http://169.254.169.254/computeMetadata/v1/"
-	metadataRecursive = "/?recursive=true&alt=json"
-	defaultTimeout    = 2 * time.Second
-)
+const programName = "GoogleAuthorizedKeysCommand"
```
Comment (dorileo) [inline google_authorized_keys/main.go:34]: I don't know why we were originally using `GoogleAuthorizedKeysCommand` here as it should be the binary name i.e. `path.Base(os.Args[0])`.

Comment (ChaitanyaKulkarni28) [reply google_authorized_keys/main.go]: Done

File: google_authorized_keys/main.go [hunk 2]
```diff
@@ -49,43 +52,14 @@ func logFormatWindows(e logger.LogEntry) string {
 	return fmt.Sprintf("%s %s: %s", now, programName, e.Message)
 }
 
-func getMetadata(key string, recurse bool) ([]byte, error) {
-	client := &http.Client{
-		Timeout: defaultTimeout,
-	}
+func getMetadata(ctx context.Context, key string) ([]byte, error) {
```
Comment (dorileo) [inline google_authorized_keys/main.go:55]: Is it worth keeping this wrapper?

**Review by dorileo — COMMENTED** [review]
Just minor comment, looking good.

File: google_authorized_keys/main.go (hunk 2 above)
Comment (ChaitanyaKulkarni28) [reply google_authorized_keys/main.go]: after refactoring it wasn't really doing much. Got rid of it. Thanks!

**Pull request merged**